cli/compose/schema/schema_test.go

```diff
@@ -65,6 +67,46 @@ func TestValidateSecretConfigNames(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestValidateSecretConfigFileRawRequired(t *testing.T) {
+	config := dict{
+		"version": "3.5",
+		"configs": dict{
+			"bar": dict{
+				"name": "foobar",
+			},
+		},
+		"secrets": dict{
+			"baz": dict{
+				"name": "foobaz",
+			},
+		},
+	}
+
+	err := Validate(config, "3.5")
+	assert.EqualError(t, err, "configs.bar file is required")
+}
+
+func TestValidateSecretConfigFileRawExclusive(t *testing.T) {
+	config := dict{
+		"version": "3.5",
+		"configs": dict{
+			"bar": dict{
+				"file": "myfile",
+				"raw":  "mydata",
+			},
+		},
+		"secrets": dict{
+			"baz": dict{
+				"file": "myfile",
+				"raw":  "mydata",
+			},
+		},
+	}
+
+	err := Validate(config, "3.5")
+	assert.EqualError(t, err, "configs.bar Must not validate the schema (not)")
```

Comment:
We need to find some way to provide a better error message here.

Reply:
It looks like these errors are handled in `cli/compose/schema/schema.go:getMostSpecificError()`. We probably need a special case for `not` that provides more detail.

Reply:
ok thanx, will check that soon